service/history/workflow/update/update_test.go

```diff
@@ -904,6 +905,81 @@ func TestWaitLifecycleStage(t *testing.T) {
 	})
 }
 
+func TestCompletedWorkflow(t *testing.T) {
+	var (
+		ctx     = context.Background()
+		effects = effect.Immediate(ctx)
+
+		meta = updatepb.Meta{UpdateId: t.Name() + "-update-id"}
+		req  = updatepb.Request{Meta: &meta, Input: &updatepb.Input{Name: t.Name()}}
+
+		store = mockEventStore{Controller: effects}
+	)
+
+	t.Run("request", func(t *testing.T) {
+		upd := update.New(meta.UpdateId)
+		err := upd.OnMessage(ctx, &req, false, store)
+		require.NoError(t, err)
+
+		status, err := upd.WaitOutcome(ctx)
+		require.NoError(t, err)
+		require.Equal(t, enumspb.UPDATE_WORKFLOW_EXECUTION_LIFECYCLE_STAGE_COMPLETED, status.Stage)
+		require.Equal(t, "Workflow Update is rejected because Workflow Execution is completed.", status.Outcome.GetFailure().GetMessage())
+		require.Equal(t, "CanceledUpdate", status.Outcome.GetFailure().GetApplicationFailureInfo().Type)
+	})
+
+	t.Run("accept", func(t *testing.T) {
```

Comment:
This test and the previous differ in the third parameter of OnMessage. In this case the workflow is running. I do not understand why we should get failure and the test name does not explain it.